src/plugin.js

```diff
@@ -1,115 +1,79 @@
-  const relativePath = screenshotDir.match(/screenshots(.*)/)[1];
-  const snapshotIdentifier = screenshotFileName.replace('.png', '');
-  const snapshotsDir = customSnapshotsDir
-    ? path.join(process.cwd(), customSnapshotsDir, relativePath)
-    : path.join(screenshotsFolder, '..', 'snapshots', relativePath);
-
-  const snapshotKebabPath = path.join(
-    snapshotsDir,
-    `${snapshotIdentifier}${kebabSnap}`
-  );
-  const snapshotDotPath = path.join(
-    snapshotsDir,
-    `${snapshotIdentifier}${dotSnap}`
-  );
-
-  const diffDir = customDiffDir
-    ? path.join(process.cwd(), customDiffDir, relativePath)
-    : path.join(snapshotsDir, '__diff_output__');
-  const diffDotPath = path.join(diffDir, `${snapshotIdentifier}${dotDiff}`);
-
-  if (fs.pathExistsSync(snapshotDotPath)) {
-    fs.copySync(snapshotDotPath, snapshotKebabPath);
-  }
-
-  snapshotResults = diffImageToSnapshot({
-    snapshotsDir,
-    receivedImageBuffer,
-    snapshotIdentifier,
-    failureThreshold,
-    failureThresholdType,
-    updateSnapshot: updateSnapshots,
-    ...options,
+import fs from 'fs-extra';
+import { COMPARE } from './constants';
+import compare from './compare';
+
+let screenshotDetails = {};
+
+const matchImageSnapshotPlugin = (name, options, config) =>
+  new Promise((resolve, reject) => {
+    // Check if we should skip the comparison, for example while debugging a case.
+    if (options.skipCompare.call()) {
+      resolve('Snapshot Compare skipped.');
+    }
+
+    // Set all paths needed in our plugin.
+    const relativePath = process.cwd();
+    const screenshotPath = screenshotDetails.path.replace(
+      config.screenshotsFolder,
+      ''
+    );
+    const screenshotFolder = screenshotDetails.path
+      .replace(relativePath, '')
+      .replace(screenshotPath, '');
+
+    const diffPath = path.join(
+      relativePath,
+      options.diffFolder || screenshotFolder,
+      screenshotPath.replace('.png', options.dotDiff)
+    );
+
+    const snapPath = path.join(
+      relativePath,
+      options.snapshotFolder || screenshotFolder,
+      screenshotPath.replace('.png', options.dotSnap)
+    );
+
+    compare(snapPath, screenshotDetails.path, diffPath, config.env.updateSnapshots || false).then(({diff, total, msg}) => {
+        if(msg) {
+          resolve(msg);
+        }
+
+        if(options.thresholdType === 'pixel') {
+            if(diff > options.threshold) {
+                reject(new Error(`Image comparison failed, the change of ${diff} is higher then the allowed ${options.threshold} pixels.`))
+            }
+        } else if(options.threshold === 'percentage') {
+            const percentage = ((total - diff) / total) * 100;
+
+            if(percentage > options.threshold) {
+                reject(new Error(`Image comparison failed, the change of ${diff} is higher then the allowed ${options.threshold} percentage.`))
+            }
+        }
+
+        resolve("Screenshot matched")
```

Comment:
Let's pick single or double quote - seems like single `'` is the standard based on previous code.